uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestCalendarFeedData.php

```diff
@@ -68,12 +74,13 @@ public function get() {
     $leaveRequests = $this->getLeaveRequests();
     $leaveRequestContacts = array_column($leaveRequests, 'contact_id');
     $contactNames = $this->getContactNames($leaveRequestContacts);
-    $absenceTypesList = $this->enabledLeaveTypesForFeed;
+
 
     $leaveData = [];
     foreach($leaveRequests as $leaveRequest) {
-      $leaveTypeName =
-        !empty($absenceTypesList[$leaveRequest['type_id']]) ? $absenceTypesList[$leaveRequest['type_id']] : 'Leave';
+      $leaveTypeName = $this->getLeaveTypeLabel($leaveRequest['type_id']);
+      //We need to adjust the time for leave request in days for display in the calendar
+      //different from the default time stored in the leave request table.
       CalendarLeaveTimeHelper::adjust($leaveRequest);
```

Comment:
It is a little bit better now, but the comment says what is this for, but not why we need to adjust things. Please explain why.